### PR TITLE
Allow KBMOD to use a different PSF per image.

### PR DESCRIPTION
**Pull request by jeremykubica**
Creates an image specific PSF within layered images and uses those for convolution, adding objects, etc.

NOTE: This is a breaking change in that it changes the API for ImageStack and LayeredImage.

### CODE REVIEW

**Comment by jeremykubica** [comment]
Closes #48

File: tests/test_image_stack.py [hunk 1]
```diff
@@ -16,7 +16,8 @@ def setUp(self):
                                            60,    # dim_y = 60 pixels,
                                            2.0,   # noise_level
                                            4.0,   # variance
-                                           2.0 * i)  # time
+                                           2.0 * i,  # time
```
Comment (maxwest-uw) [inline tests/test_image_stack.py:19]: should we have a test added for the image_stack class to check that it can properly access individual image psfs?

**Review by maxwest-uw — COMMENTED** [review]
altogether these are great changes, I had one suggestion for adding a small test to ensure that image_stack can access individual image psfs.
one thing to consider, is there anything we need to do/anybody we need to warn that this is 'breaking' change since it changes the image api? I know that kbmod isn't part of any pipelines yet but I want to make sure we don't break someones ongoing work!

File: tests/test_image_stack.py (hunk 1 above)
Comment (jeremykubica) [reply tests/test_image_stack.py:19]: Good call. I added a new test (test_different_psfs) to check that it is handling the per-image PSFs correctly.

**Review by maxwest-uw — APPROVED** [review]
everything looks good to me! thanks for adding that test!

**Review by maxwest-uw — APPROVED** [review]
👍

**Pull request merged**